dali/test/python/checkpointing/test_dali_checkpointing.py

```diff
@@ -287,6 +288,90 @@ def pipeline():
     compare_pipelines(p, restored, 1, 20)
 
 
+@dataclass
+class BaseDecoderConfig:
+    shard_id: int
+    num_shards: int
+    stick_to_shard: bool
+    pad_last_batch: bool
+    random_shuffle: bool
+
+@dataclass
+class VideoConfig:
+    sequence_length : int
+    stride : int
+    step : int
+
+
+@cartesian_params(
+    (0, 1, 3),
+    (1, 3),
+    (0, 2),
+    (
+        BaseDecoderConfig(shard_id=0, num_shards=1, stick_to_shard=True, pad_last_batch=True, 
+                          random_shuffle=True),
+        BaseDecoderConfig(shard_id=4, num_shards=7, stick_to_shard=True, pad_last_batch=True, 
+                          random_shuffle=False),
+        BaseDecoderConfig(shard_id=6, num_shards=7, stick_to_shard=False, pad_last_batch=False, 
+                          random_shuffle=False),
+        BaseDecoderConfig(shard_id=0, num_shards=2, stick_to_shard=False, pad_last_batch=False, 
+                          random_shuffle=True),
+    ),
+    (
+        VideoConfig(sequence_length=3, stride=1, step=-1),
+        VideoConfig(sequence_length=3, stride=1, step=5),
+    ),
+)
+def test_video_reader(num_epochs, batch_size, iters_into_epoch,
+                      config: BaseDecoderConfig, video: VideoConfig):
+
+    files = 2 * [os.path.join(get_dali_extra_path(), 'db/video/multiple_framerate/10/10fps.mp4')]
```

Comment:
I wonder if that is enough data? I.e. that the multiple shards cases don't get too trivial? Can we use all the files from `multiple_framerate` at once @awolant?

Reply:
I checked that at some point but it might be good to add some assert in the test to check that there are at least 2 samples in each shard and that the requested number of iterations in epoch is possible

Reply:
It turns out that some testcases indeed were trivial, thank you! I added assets and changed the video files used to fix that